src/main/java/me/jellysquid/mods/lithium/common/entity/LithiumEntityCollisions.java

```diff
@@ -251,4 +258,29 @@ public static boolean isBoxFullyWithinWorldBorder(WorldBorder border, Box box) {
 
         return shapes;
     }
+
+
+    private static final Predicate<CollisionBoxOverridingEntity> EXCEPT_SPECTATOR = collisionBoxOverridingEntity -> !((Entity) collisionBoxOverridingEntity).isSpectator();
+
+    /**
+     * Partial [VanillaCopy] Classes overriding Entity.getHardCollisionBox(Entity other) or Entity.getCollisionBox()
+     * @param entityView the world
+     * @param selection the box the entities have to collide with
+     * @param entity the entity that is searching for the colliding entities
+     * @return list of entities with collision boxes
+     */
+    public static List<Entity> getEntitiesWithCollisionBoxForEntity(EntityView entityView, Box selection, Entity entity) {
+        if (entity instanceof BoatEntity || entity instanceof StriderEntity || entity instanceof MinecartEntity || !(entityView instanceof World)) {
```

Comment:
`collision_only_boat_shulker` looks particularly fragile here. I don't feel comfortable adding it on the basis that it relies on having assumed knowledge on what entities may or may not do in the game.

Reply:
I agree it is very fragile with other mods, the alternative would be to check for each entity type whether it overrides getCollisionBox or getHardCollisionBox at runtime using reflection and cache the result.  I have the code available for that, but it probably requires mixining into the Entity constructor, as I don't know another way to discover all (used) entity classes during runtime